src/main/java/com/hotels/mutantswarm/MutantSwarmTestExtension.java

```diff
@@ -0,0 +1,337 @@
+    Preconditions.checkState(Files.exists(file), "File " + file + " does not exist");
+  }
+
+  //These methods are the ones to set up the tests, they are copy pasted from HiveRunner
+  //TODO: find a way to access these methods from HiveRunner instead of copying everything
+  private HiveShellContainer createHiveServerContainer(List<? extends Script> scripts, Object testCase, Path basedir)
+      throws IOException {
+    return core.createHiveServerContainer(scripts, testCase, basedir, config);
+  }
+
+  private void setupConfig(Object target) {
+    Set<Field> fields = ReflectionUtils.getAllFields(target.getClass(),
+        Predicates.and(
+            withAnnotation(HiveRunnerSetup.class),
+            withType(HiveRunnerConfig.class)));
+
+    Preconditions.checkState(fields.size() <= 1,
+        "Only one field of type HiveRunnerConfig should be annotated with @HiveRunnerSetup");
+
+    if (!fields.isEmpty()) {
+      config.override(ReflectionUtils
+          .getFieldValue(target, fields.iterator().next().getName(), HiveRunnerConfig.class));
+    }
+  }
+
+  private void tearDown(Object target) {
+    if (container != null) {
+      log.info("Tearing down {}", target.getClass());
+      container.tearDown();
+    }
+    deleteTempFolder(basedir);
+  }
+
+  private void deleteTempFolder(Path directory) {
+    try {
+      FileUtils.deleteDirectory(directory.toFile());
+    } catch (IOException e) {
+      log.debug("Temporary folder was not deleted successfully: " + directory);
+    }
+  }
+
+  @Override
+  public void afterEach(ExtensionContext extensionContext) {
+    tearDown(extensionContext.getRequiredTestInstance());
+  }
+
+  //These are all the methods used to generate a swarm
+  //TODO: Add this in to a core class and avoid repeating code here and in the MutantSwarmRule
+  private MutantSwarmSource setUpScripts() {
+    log.debug("Setting up scripts");
+    List<MutantSwarmScript> scripts = new ArrayList<>();
+
+    MutantSwarmStatement.Factory statementFactory = new MutantSwarmStatement.Factory();
+
+    for (int i = 0; i < scriptsUnderTest.size(); i++) {
+      Script testScript = scriptsUnderTest.get(i);
+      emulator = HiveRunnerConfig.getCommandShellEmulator();
+
+      List<Statement> scriptStatements = new StatementSplitter(emulator).split(testScript.getSql());
+
+      List<MutantSwarmStatement> statements = new ArrayList<>();
+      for (int j = 0; j < scriptStatements.size(); j++) {
+        String statementText = scriptStatements.get(j).getSql();
+        MutantSwarmStatement statement = statementFactory.newInstance(i, j, statementText);
+        statements.add(statement);
+      }
+      MutantSwarmScript script = new MutantSwarmScript.Impl(i, testScript.getPath(), statements);
+      scripts.add(script);
+    }
+    return new MutantSwarmSource.Impl(scripts);
+  }
+
+  private Swarm generateSwarm() {
+    log.debug("Setting up mutants");
+    MutantSwarmSource source = setUpScripts();
```

Comment:
Move `MutantSwarmSource setUpScripts()` from HiveRunnerRule into `HiveRunnerCore` and make `protected` and change signature to `MutantSwarmSource setUpScripts(List<? extends Script> scriptsUnderTest)` so we can pass in the scripts from both the rule and extension.